### DISCUSSION TITLE
Show context menu programatically?

### DISCUSSION THREAD

**Discussion by ptdev** [HavenDV/H.NotifyIcon#154] [Q&A]
Hey,

First of all, great job with this lib! 
Now for my question: Is it possible to open the context menu programatically?

**Comment by HavenDV** [comment]
Hello. This is possible, but unfortunately there is currently no public method for this.
You can wait or do it yourself by providing a PR

**Reply by ptdev** [reply]
Hey, thanks for the reply.

I took a closer look at the code and tried changing the visibility of the ShowContextMenu from private to public and that (sort of) worked.

The method still requires a cursorposition to be supplied and that requires some extra steps but it works.
(having the cursorposition default to a position like the middle of the icon or something when null would be awesome, but couldn't figure out how to do that).

I would much rather have this feature on the official NuGet package so that I don't have to keep a local copy so if you're fine with including this feature on some future version that would be greatly appreciated! :heart:

And I can or course wait or, if you wish I can also submit a PR. Just seemed a bit weird submitting one just to change the visibility which is 1 word 😅 

Thanks again, cheers!

**Reply by HavenDV** [reply]
https://github.com/HavenDV/H.NotifyIcon/blob/efd1a417a7d7a8fea62d1a2b4c3bcd760bcde6b1/src/libs/H.NotifyIcon/Core/TrayInfo.cs
There is a method here, `TrayInfo.GetTrayLocation(int space = 2)`, which can possibly be used as a default value, but I’m not sure if this is correct to take Desktop Scale and DPI into account
I'd be grateful if you could test this

**Reply by HavenDV** [reply]
https://github.com/HavenDV/H.NotifyIcon/commit/41b70d03610c1f9c64ab7fad0a9aa2ef8c8fa12b
New version with public methods should be available soon

**Reply by ptdev** [reply]
Thanks!

I tried using that GetTrayLocation method as you suggested, but it's apparently giving me wrong values.

I'm on a 2560x1440 display and it returns {X=2560,Y=1378} which is not where the icon is but way too much to the right (looks like it's sort of where the windows notifications show up maybe?)

But either way, just having the method being public already solves my core issue, so thanks a lot for the quick support!

Cheers!👍
